eveoh/room-display
# Replace Gulp by Webpack

Ref. eveoh/tasks#136

## Proposed changes

Replaces Gulp by Webpack. Also applies some ES6 features (replace `var` by `let`/`const`, JS import style..) because Webpack allows that. Gulp didn't.

Moved JS from `error.html` to a separate `error.js` file.

The `gulp_war` task which produced a .war file hasn't been recreated in Webpack, because we don't do custom Tomcat deployment stuff anymore. 

Gradle has been removed from the project, because we aren't doing anything with Java in here. The publishing is now being performed with npm/yarn thanks to the `mvn-up` package.

Reorganized the project structure removing the `WEB-INF/web.xml` together with `main` and `resources` dirs. This conform to discussions we had while Webpacking `echo-front`.

Added GitHub actions for branch, PR and publishing.

Fixed a bug where the page kept reloading every 60ms when `config.json` file has not been found. It's now set to reload every 60 seconds when the error occurs.

## Testing

- Run `yarn build` and `yarn dist:tar` tasks successfully.
    - verify that the `manifest.appcache` has its `@version@` and `@timestamp@` placeholders replaced by the appropriate values.
- Run `yarn start` task successfully and use the room-display app:
    - Provision a room and view it;
    - Navigate to `/settings` and configure and view various data source `2015` rooms;
    - Navigate to `/error` whether the error page works;
    - Add a `override.json` file to see whether it's being picked up and overrides the default `config.json` configs.
    - Verify `manifest.appcache` file caching.

- Manually test publish to our snapshots Maven repo.
    - Download the tar, untar it and verify content.

## Documentation

Has been taken care of in README.

## Upgrade notes for breaking changes

- [x] `UPGRADE_NOTES.md` has been updated (when applicable).
